mitodl/open-discussions
# Add indexing user as first moderator to every channel if not already a moderator

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #1369

#### What's this PR do?
- Creates every channel as the indexing user, then adds the user who is submitting the new channel as a moderator for that channel.
- Adds a management command `add_index_user_to_channels` that will iterate over every `Channel` object and insert the indexing user into the moderators list if not already present.

#### How should this be manually tested?
- On the master branch, while logged in as anyone besides the indexing user, create some channels at `/manage/c/new`.  Add another user or two as moderators to some of the channels.
- Switch to this branch and run `docker-compose run web python manage.py add_index_user_to_channels`.  You should see a bunch of messages like `Channel <name> SUCCESS: Indexing user added or already present`
- There may be errors if you have `Channel` objects that don't actually exist in reddit, or channels that have 0 moderators in reddit.  I don't think there's any way to fix those.
- Every channel where the indexing user was missing should now have the indexing user as the most senior moderator, and all the other moderators should still be present in their original order.

#### Where should the reviewer start?
[channels.management.commands.add_index_user_to_channels.py](https://github.com/mitodl/open-discussions/pull/1409/files#diff-4943cf61ee09cb70ba70d15b18b56884)

[channels.serializers.ChannelSerializer.create](https://github.com/mitodl/open-discussions/pull/1409/files#diff-7e5cbd3bde192b46904dbf56f630dc13R137)
